scapy/contrib/concox.py

```diff
@@ -0,0 +1,314 @@
+                    0x06: "Over speed",
+                    0x09: "Displacement",
+                    0x0a: "Enter GPS dead zone",
+                    0x0b: "Exit GPS dead zone",
+                    0x0c: "Power on",
+                    0x0d: "GPS First fix notice",
+                    0x0e: "Low battery",
+                    0x0f: "Low battery protection",
+                    0x10: "SIM Change",
+                    0x11: "Power off",
+                    0x12: "Airplane mode",
+                    0x13: "Disassemble",
+                    0x14: "Door",
+                    0xfe: "ACC On",
+                    0xff: "ACC Off",
+                }), lambda pkt: len(pkt.original) > 5 and pkt.protocol_number
+            in (0x13, 0x15, 0x16)),
+        ConditionalField(
+            ByteEnumField("language", 0x00,
+                          LANGUAGE), lambda pkt: len(pkt.original) > 5 and pkt.
+            protocol_number in (0x13, 0x15, 0x16)),
+        # Information transmission
+        ConditionalField(
+            ByteEnumField("subprotocol_number", 0x00,
+                          SUBPROTOCOL_NUMBERS), lambda pkt: len(pkt.original) >
+            5 and pkt.protocol_number in (0x94, )),
+        ConditionalField(
+            ShortField('external_battery',
+                       0x00), lambda pkt: len(pkt.original) > 5 and pkt.
+            protocol_number in (0x94, ) and pkt.subprotocol_number == 0x00),
+        ConditionalField(
+            FlagsField('external_io_detection', 0x00, 8, [
+                'door_status',
+                'trigger_status',
+                'io_status',
+            ]), lambda pkt: len(pkt.original) > 5 and pkt.protocol_number in (
+                0x94, ) and pkt.subprotocol_number == 0x05),
+        # Default
+        XShortField('information_serial_number', None),
+        XShortField('crc', None),
+    ]
+
+
+class CRX1New(Packet):
+    name = "CRX1 New"
+    fields_desc = [
+        XShortField('start_bit', 0x7878),
+        ConditionalField(ByteField(
+            'default_packet_length',
+            None,
+        ), lambda pkt: pkt.start_bit == 0x7878),
+        ConditionalField(ShortField(
+            'extended_packet_length',
+            None,
+        ), lambda pkt: pkt.start_bit == 0x7979),
+        ConditionalField(
+            PacketLenField('default_packet_content',
+                           None,
+                           CRX1NewPacketContent,
+                           length_from=lambda pkt: pkt.default_packet_length),
+            lambda pkt: pkt.start_bit == 0x7878),
+        ConditionalField(
+            PacketLenField('extended_packet_content',
+                           None,
+                           CRX1NewPacketContent,
+                           length_from=lambda pkt: pkt.extended_packet_length),
+            lambda pkt: pkt.start_bit == 0x7979),
+        XShortField('end_bit', 0x0d0a),
+    ]
+
+
+bind_layers(TCP, CRX1New)
+bind_layers(UDP, CRX1New)
```

Comment:
This will break Scapy. You should have a port, or something to chose between this layer and others

Reply:
According to Google the port is 8821.

Reply:
Actually, you can configure any port in the device

Reply:
The manufacturer does not specify a default port in the protocol or in the manual.

Reply:
Binding literally all UDP and TCP packets to CRX doesn't seem like a viable solution either

Reply:
Of course, it's just to keep the register that the set port is unofficial.